lib/services/api/api_service.ts

```diff
@@ -58,8 +61,11 @@ export class SNApiService extends PureService {
     return this.host;
   }
 
-  public setSession(session: Session) {
+  public async setSession(session: Session, fromDisk: boolean = false) {
```

Comment:
It would be more clear if `fromDisk` was renamed to `persist`. Because it's not clear how fromDisk will be used from a caller's perspective. `persist` makes this much more clear. Then if(persist) {save to disk}.